src/sniffer.c

```diff
@@ -2155,701 +2189,861 @@ static void ShowTlsSecrets(SnifferSession* session)
 }
 #endif /* SHOW_SECRETS */
 
+typedef struct {
+    int type;
+    union {
+    #ifndef NO_RSA
+        RsaKey rsa;
+    #endif
+    #if !defined(NO_DH) && defined(WOLFSSL_DH_EXTRA)
+        struct {
+            word32 pLen; /* modulus length */
+            word32 privKeySz;
+            byte   privKey[52]; /* max for TLS */
+            DhKey  key;
+        } dh;
+    #endif
+    #ifdef HAVE_ECC
+        ecc_key ecc;
+    #endif
+    #ifdef HAVE_CURVE25519
+        curve25519_key x25519;
+    #endif
+    #ifdef HAVE_CURVE448
+        curve448_key x448;
+    #endif
+    } priv;
 
-/* Process Keys */
+    union {
+    #ifdef HAVE_ECC
+        ecc_key ecc;
+    #endif
+    #ifdef HAVE_CURVE25519
+        curve25519_key x25519;
+    #endif
+    #ifdef HAVE_CURVE448
+        curve448_key x448;
+    #endif
+    } pub;
```

Comment:
How come no DH and no RSA? 

Also, if ecc, x25519 and x448 are disabled at configure time, then you have an empty union.  Is that valid?